AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -154,16 +154,31 @@ extension Statement: StatementModelConvertible {
 
     private func convertCollection(field: ModelField, schema: ModelSchema, from element: Element, path: [String]) -> Any? {
         if field.isArray && field.hasAssociation,
-           case let .some(.hasMany(associatedFieldName: associatedFieldName)) = field.association
+           case let .some(.hasMany(associatedFieldName: associatedFieldName, targetNames: targetNames)) = field.association
         {
-            let primeryKeyName = schema.primaryKey.isCompositeKey
-                ? ModelIdentifierFormat.Custom.sqlColumnName
-                : schema.primaryKey.fields.first.flatMap { $0.name }
-            let primeryKeyValue = primeryKeyName.flatMap { getValue(from: element, by: path + [$0]) }
-
-            return primeryKeyValue.map {
-                DataStoreListDecoder.lazyInit(associatedId: String(describing: $0), associatedWith: associatedFieldName)
+            // Construct the lazy list based on the field reference name and `@@primarykey` or primary key field of the parent
+            if targetNames.count <= 1, let associatedFieldName = associatedFieldName {
+                let primaryKeyName = schema.primaryKey.isCompositeKey
+                    ? ModelIdentifierFormat.Custom.sqlColumnName
+                    : schema.primaryKey.fields.first.flatMap { $0.name }
+                let primaryKeyValue = primaryKeyName.flatMap { getValue(from: element, by: path + [$0]) }
+
+                return primaryKeyValue.map {
+                    DataStoreListDecoder.lazyInit(associatedIds: [String(describing: $0)],
+                                                  associatedWith: [associatedFieldName])
+                }
+            } else {
+                // If `targetNames` is > 1, then this is a uni-directional has-many, thus no reference field on the child
+                // Construct the lazy list based on the primary key values and the corresponding target names
+                let primaryKeyNames = schema.primaryKey.fields.map { $0.name }
+                let primaryKeyValues = primaryKeyNames
+                    .map { getValue(from: element, by: path + [$0]) }
+                    .compactMap { $0 }
+                    .map { String(describing: $0) }
+                return DataStoreListDecoder.lazyInit(associatedIds: primaryKeyValues,
+                                                     associatedWith: targetNames)
```

Comment:
this is the functional change to pass down the `targetNames` for lazy loading the uni-directional has-many use case. We have to extract out the values of the primary key values from the parent model, instead of just using the `@@primaryKey` field value.

Reply:
Yes I believe so, they are the foreign keys fields of the parent model on the child model, which is an array due the parent model containing a composite key. For example, on the Post schema, the Post has a has-many reference to the Comments
```
      .hasMany(post4.comments, is: .optional, ofType: Comment4.self, associatedWith: Comment4.keys.post4CommentsPostId, targetNames: ["post4CommentsPostId", "post4CommentsTitle"]),
```

The `targetNames` represent the "associated with fields". On the Comment model,
```
struct Comment {
  var post4CommentsPostId: String
  var post4CommentsTitle: String
}
```

Since the comment does not have a reference object to Post, there is no "post: Post" field on Comment, and the foreign key composite keys are just strings. The proposed change is to add "targetNames" to the parent model's hasMany fields to point to "post4CommentsPostId" and "post4CommentsTitle"